tools/tsdb-series/main.go

```diff
@@ -130,19 +168,50 @@ func printBlockIndex(ctx context.Context, blockDir string, printChunks bool, ser
 			continue
 		}
 
+		output := seriesWithStats{}
+		output.Series = lbls.Map()
+
 		if seriesStats {
-			minT, maxT, samples := computeChunkStats(chr, chks)
-			dpm := float64(samples) / (maxT.Sub(minT).Minutes())
-			fmt.Println("series:", lbls.String(), "minT:", formatTime(minT), "maxT:", formatTime(maxT), "samples:", samples, "dpm:", dpm)
-		} else {
-			fmt.Println("series:", lbls.String())
+			output.MinTime, output.MaxTime, output.Samples = computeChunkStats(chr, chks, minTime.UnixMilli(), maxTime.UnixMilli())
+			output.DPM = float64(output.Samples) / (time.Millisecond * time.Duration(output.MaxTime-output.MinTime)).Minutes()
+			if math.IsNaN(output.DPM) || math.IsInf(output.DPM, 0) {
+				output.DPM = 0
+			}
 		}
 
 		if printChunks {
 			for _, c := range chks {
-				fmt.Println("chunk:", c.Ref,
-					"min time:", c.MinTime, formatTime(timestamp.Time(c.MinTime)),
-					"max time:", c.MaxTime, formatTime(timestamp.Time(c.MaxTime)))
+				output.Chunks = append(output.Chunks, chunk{
+					Ref:     c.Ref,
+					MinTime: c.MinTime,
+					MaxTime: c.MaxTime,
+				})
+			}
+		}
+
+		if jsonOutput {
```

Comment:
_this a nitpick on a maybe throwaway code, so feel free to ignore_, but i'd maybe pull out the printing logic into something like a `printer` which can be implemented by an `stdoutPrinter` and a `jsonPrinter` to keep `printBlockIndex` shorter.